src/idl_parser.cpp

```diff
@@ -2004,18 +2042,31 @@ CheckedError Parser::DoParse(const char *source, const char **include_paths,
   while (token_ != kTokenEof) {
     if (opts.proto_mode) {
       ECHECK(ParseProtoDecl());
-    } else if (token_ == kTokenNameSpace) {
-      ECHECK(ParseNamespace());
-    } else if (token_ == '{') {
-      if (!root_struct_def_)
-        return Error("no root type set to parse json with");
-      if (builder_.GetSize()) {
-        return Error("cannot have more than one json object in a file");
+  }
+  else if (token_ == kTokenNameSpace) {
+    ECHECK(ParseNamespace());
+  } else if (token_ == '{' || token_ == '[') {
```

Comment:
OK I took it out. I want Flatbuffer to be my one-stop parsing solution for all the json I have to deal with in that project I am working. Some json files I have no control over are in that straight top level array format so I will just keep the change around in a fork.